tests/ctst/steps/bucket-policies/common.ts

```diff
@@ -0,0 +1,475 @@
+    const action = this.getSaved<ActionPermissionsType>('currentAction');
+    let effect = AuthorizationType.DENY;
+    let resources;
+    const applies = doesApply === 'applies';
+    const bucketName = this.getSaved<string>('bucketName');
+    if (doesExists === 'existing') {
+        if (applies) {
+            if (isAllow === 'ALLOW') {
+                authzConfiguration.Resource = AuthorizationType.ALLOW;
+                effect = AuthorizationType.ALLOW;
+            } else {
+                authzConfiguration.Resource = AuthorizationType.DENY;
+                effect = AuthorizationType.DENY;
+            }
+            resources = [
+                `arn:aws:s3:::${bucketName}`,
+                `arn:aws:s3:::${bucketName}/*`,
+            ];
+        } else {
+            authzConfiguration.Resource = AuthorizationType.IMPLICIT_DENY;
+            // Effect is ALLOW on purpose, to ensure we properly handle implicit denies
+            effect = AuthorizationType.ALLOW;
+            resources = [
+                `arn:aws:s3:::${bucketName}`,
+                `arn:aws:s3:::${bucketName}/*`,
+            ];
+        }
+        if (action.excludePermissionOnBucketObjects) {
+            resources.pop();
+        }
+        if (action.excludePermissionsOnBucket) {
+            resources.shift();
+        }
+    } else {
+        authzConfiguration.Resource = AuthorizationType.NO_RESOURCE;
+        return;
+    }
+    const expectFromConditions = this.getSaved<AuthorizationType>('expectFromConditions');
+    if (expectFromConditions) {
+        authzConfiguration.Resource = expectFromConditions;
+    }
+    this.addToSaved('authzConfiguration', authzConfiguration);
+    const currentIdentityArn = this.getSaved<string>('identityArn');
+    let principal = currentIdentityArn;
+    const identityType = this.getSaved<string>('identityType') as EntityType;
+    if (identityType === EntityType.ASSUME_ROLE_USER
+        || identityType === EntityType.ASSUME_ROLE_USER_CROSS_ACCOUNT
+        || identityType === EntityType.DATA_CONSUMER) {
+        principal = '*';
+    }
+    if (!applies) {
+        principal = `${currentIdentityArn}badname`;
+    }
+    const basePolicy = {
+        Version: '2012-10-17',
+        Statement: [
+            {
+                Effect: effect,
+                Action: action.permissions,
+                Resource: resources,
+                Principal: {
+                    AWS: principal,
+                },
+                Condition: {},
+            },
+        ],
+    };
+    const conditionForPolicy = this.getSaved('conditionForPolicy');
+    if (conditionForPolicy) {
+        basePolicy.Statement[0].Condition = conditionForPolicy;
+    }
+    const result = await S3.putBucketPolicy({
+        bucket: this.getSaved<string>('bucketName'),
+        policy: JSON.stringify(basePolicy),
+    });
+    assert.ifError(result.stderr || result.err);
+});
```

Comment:
Referencing the whole function: is it possible to have a similar setup for IAM policy conditions as well? We will soon need such a structure to tes the aws:secureTransport condition in IAM policies like [here](https://github.com/scality/Integration/pull/1247/files#diff-6f742aeecde49bbbbf6e26b9f804790091664c62f0d95792c5a54da1462edcbeR29)

Reply:
Yes, we have it already here: https://github.com/scality/Zenko/pull/2023/files#diff-d3226e11b069c1aeb67b94370266647a291d7a78f76f3dc9b176a533b41d76e9R61